runtime/common/src/lib.rs

```diff
@@ -420,6 +425,46 @@ where
 	}
 }
 
+/// XcmMessageHandler of `DmpMessageHandler` and `XcmpMessageHandler` implementations.
+/// if xcm is paused, the `max_weight` of each handle method is set to `0`.
+///
+/// Parameters type:
+/// - `A`: `DmpMessageHandler` or `XcmpMessageHandler`
+pub struct XcmMessageHandlerOrPaused<Runtime, A>(PhantomData<(Runtime, A)>);
```

Comment:
A better name suggest? XcmMessageHandlerOr**Queued** or just XcmMessageHandler**s**?

Reply:
`XcmMessageHandler` should be fine.